### PR TITLE
Template/certificate preview CEMS-2108

### PR DESCRIPTION
**Pull request by steve-giles**
This update provides the ability to preview a certificate using a document template and data for a template.

Updates:
- UI provided with a "preview" button, and input filed for data key
- Frontend service makes a call to the dts render endpoint
- Modal window to display certificate preview

For walkthrough and testing instructions see this three minute video: https://drive.google.com/file/d/1-uRzjrLpYsDJHCUkrAWVApArmNyx_vAw/view?usp=sharing

### CODE REVIEW

File: src/app/dts/dts.service.ts [hunk 1]
```diff
@@ -58,6 +58,18 @@ export class DtsService {
     return this.http.get(url, {responseType: 'text'});
   }
 
+  /**
+   * Retrieves a rendered template as a certificate populated with data
+   * @param documentType currently limited to 'enrollment'
+   * @param templateKey unique template identifier
+   * @param dataKey unique data identifier
+   */
+  public renderTemplate(documentType: string, templateKey: string, dataKey: string): Observable<string> {
```
Comment (alcfeoh) [inline src/app/dts/dts.service.ts:67]: Just a tip if you ever have more documentTypes than just ` 'enrollment'`, you could create a type definition using union types like so, which is a lot more specific than `string` while still using strings everywhere:
```export type DocumentType = 'enrollment' | 'somethingElse';```

File: src/app/dts/template-editor/template-editor.component.ts [hunk 2]
```diff
@@ -58,7 +58,8 @@ export class TemplateEditorComponent implements OnInit {
       templateData: new FormControl(),
       templateKey: new FormControl(),
       templateId: new FormControl(),
-      author: new FormControl()
+      author: new FormControl(),
```
Comment (alcfeoh) [inline src/app/dts/template-editor/template-editor.component.ts]: Note: `formBuilder.group` accepts a shorter syntax for this:
```
this.formBuilder.group({
      templateData: '',	  
      templateKey: '',
      templateId: '',	     
      author: ''	
});
```

**Review by alcfeoh — APPROVED** [review]
All looks good, I added a couple of suggestions for the future but nothing very important for now.

File: src/app/dts/template-editor/template-editor.component.ts (hunk 2 above)
Comment (steve-giles) [reply src/app/dts/template-editor/template-editor.component.ts]: Applied the recommended change.

File: src/app/dts/dts.service.ts (hunk 1 above)
Comment (steve-giles) [reply src/app/dts/dts.service.ts:67]: Thank you for the tip.  We can apply change this once we're supporting more document types.

**Pull request merged**